shotgun_api3/shotgun.py

```diff
@@ -2587,7 +2587,10 @@ def _upload_to_sg(self, entity_type, entity_id, path, field_name, display_name,
             url = urllib.parse.urlunparse((self.config.scheme, self.config.server,
                                            "/upload/upload_file", None, None, None))
             if display_name is None:
-                display_name = os.path.basename(path)
+                if six.PY2:
+                    display_name = os.path.basename(path)
+                else:
+                    display_name = os.path.basename(path.decode("utf-8"))
```

Comment:
Can you 15 lines above. There is a big comment about unicode path and we encode path in utf-8... Looks like the opposite of what's done here. We might want to refactor/clean/review all that